src/sinks/apex/mod.rs

```diff
@@ -0,0 +1,170 @@
+use bytes::Bytes;
+use futures::FutureExt;
+use futures_util::SinkExt;
+use http::{Request, StatusCode, Uri};
+use hyper::Body;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+#[cfg(feature = "apex-integration-tests")]
+#[cfg(test)]
+mod integration_tests;
+
+use crate::{
+    config::{
+        AcknowledgementsConfig, GenerateConfig, Input, SinkConfig, SinkContext, SinkDescription,
+    },
+    event::Event,
+    http::HttpClient,
+    sinks::util::{
+        encoding::Transformer,
+        http::{BatchedHttpSink, HttpEventEncoder, HttpSink},
+        BatchConfig, BoxedRawValue, JsonArrayBuffer, SinkBatchSettings, TowerRequestConfig,
+        UriSerde,
+    },
+};
+
+#[derive(Deserialize, Serialize, Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct ApexSinkConfig {
+    uri: UriSerde,
+    project_id: String,
+    api_token: String,
+
+    #[serde(default)]
+    batch: BatchConfig<ApexDefaultBatchSettings>,
+
+    #[serde(default)]
+    request: TowerRequestConfig,
+
+    #[serde(
+        default,
+        deserialize_with = "crate::serde::bool_or_struct",
+        skip_serializing_if = "crate::serde::skip_serializing_if_default"
+    )]
+    acknowledgements: AcknowledgementsConfig,
+}
```

Comment:
Apologizes for this, but based on work that was recently merged to `master`, this will require some additional changes related to the on-going [configuration schema](https://github.com/vectordotdev/vector/issues/12141) work. You'll also likely want to merge the latest `master` before doing anything that I mention below.

You can take a look at https://github.com/vectordotdev/vector/blob/master/src/sinks/http.rs#L48 and crib example code/snippets from that.  Essentially, copy the macros on the `HttpSinkConfig` itself, add doc comments for `uri`, `project_id`, and `api_token` that match their Cue-based documentation, and the rest of the fields should end up with a `#[configurable(derived)]` helper attribute.

Reply:
No worries! I'll make the necessary updates